chromium_src/net/url_request/url_request_http_job.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "net/url_request/url_request_http_job.h"
+
+#include "net/base/features.h"
+#include "net/cookies/cookie_monster.h"
+
+namespace {
+
+bool ShouldUseEphemeralStorage(net::URLRequestHttpJob* http_job) {
+  if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
+    return false;
+
+  const net::IsolationInfo& isolation_info =
+      http_job->request()->isolation_info();
+  if (!isolation_info.top_frame_origin().has_value() ||
+      !isolation_info.frame_origin().has_value())
+    return false;
+  if (*isolation_info.top_frame_origin() == *isolation_info.frame_origin())
+    return false;
+
+  return true;
+}
+
+}  // namespace
+
+#define BRAVE_ADDCOOKIEHEADERANDSTART                                          \
+  if (ShouldUseEphemeralStorage(this)) {                                       \
+    DCHECK(request()->isolation_info().top_frame_origin().has_value());        \
```

Comment:
I think it would be better to move this DCHECK here https://github.com/brave/brave-core/pull/7387/files#diff-2b849c37882ef112e7d503ffa24e13c9987f94b8f4cc5ee2870c4f87a357b8c4R24 so it doesn't have to be duplicated and also so you don't need the NOLINT for the weird parens

Reply:
I can remove the DCHECK, but:
 1. At line 24 we should have returned early if request()->isolation_info().top_frame_origin().has_value() so the assertion might be a bit redundant.
 2. I *think* that since the code in the first if-block spans multiple lines it still needs braces and the NOLINT continues to be unavoidable.

Reply:
braces are only needed if the there is more than one statement

Reply:
Okay. I've removed the DCHECK and the brace. Now the NOLINT statement is unnecessary.